cargo-shuttle/src/client.rs

```diff
@@ -174,8 +174,18 @@ impl Client {
         self.delete(path).await
     }
 
-    pub async fn get_logs(&self, project: &str, deployment_id: &Uuid) -> Result<Vec<LogItem>> {
-        let path = format!("/projects/{project}/deployments/{deployment_id}/logs");
+    pub async fn get_logs(
+        &self,
+        project: &str,
+        deployment_id: &Uuid,
+        mode: &str,
+        len: u32,
+    ) -> Result<Vec<LogItem>> {
+        let path = match mode {
+            "head" => format!("/projects/{project}/deployments/{deployment_id}/logs?head={len}"),
+            "tail" => format!("/projects/{project}/deployments/{deployment_id}/logs?tail={len}"),
+            _ => format!("/projects/{project}/deployments/{deployment_id}/logs"),
+        };
```

Comment:
You could use something like this here instead: https://stackoverflow.com/a/30154791. We should also consider creating an enum for `mode` and putting it in common, so we can use it everywhere we need it.